File: 18주차/KM/1013-partition-array-into-three-parts-with-equal-sum.java
class Solution {
    public boolean canThreePartsEqualSum(int[] arr) {
        int target = 0;
        for(int a : arr){
            target += a;
        }
        if(target%3 != 0) return false;
        target = target / 3;
        int sum = 0;
        int flag = 3;
        for(int i = 0; i<arr.length; i++){
            sum += arr[i];
            if(sum == target){
                sum = 0;
                flag--;
            }
        }
        return (flag<=0) ? true : false;
    }
    
}

File: 18주차/DE/1013. Partition Array Into Three Parts With Equal Sum.js
/**
 * @param {number[]} arr
 * @return {boolean}
 */
 function canThreePartsEqualSum(arr) {
    let num = arr.reduce((a, b) => a + b) / 3;
    let cnt = 0;
    let sum = 0;
    
    for (let n of arr) {
        sum += n;
        if (sum == num) {
            cnt++;
            sum = 0;
        }
    }
    
    return cnt >= 3;
};